cwms-data-api/src/main/java/cwms/cda/security/OpenIDAccessManager.java

```diff
@@ -69,15 +72,26 @@ public void manage(Handler handler, @NotNull Context ctx, @NotNull Set<RouteRole
     private DataApiPrincipal getUserFromToken(Context ctx) throws CwmsAuthException {
         try {
             Jws<Claims> token = jwtParser.parseClaimsJws(getToken(ctx));
-            String username = token.getBody().get("preferred_username",String.class);
+            Claims claims = token.getBody();
+            final String issuer = claims.getIssuer();
+            final String subject = claims.getSubject();
+            final String username = issuer + "::" + subject;
```

Comment:
is there any risk of issuer or subject already containing "::" ?  would it matter if they did?

Reply:
I don't think so. :: wouldn't be valid in a URL, and subject is either a UUID or some form of username. It's possible it could be an issue in the future, but it wouldn't be difficult to change later. Long term I want them to be their own columns, but we're kinda working on sort timelines so this particular form is more of a stop gap.